signer/src/stacks/contracts.rs

```diff
@@ -38,5 +52,79 @@ pub trait AsContractCall {
     fn post_conditions(&self, deployer: StacksAddress) -> StacksTxPostConditions;
 }
 
+/// This struct is used to generate a properly formatted Stacks transaction
+/// for calling the complete-deposit-wrapper function in the sbtc-deposit
+/// smart contract.
+#[derive(Copy, Clone, Debug)]
+pub struct CompleteDepositV1 {
+    /// The outpoint of the bitcoin UTXO that was spent as a deposit for
+    /// sBTC.
+    pub outpoint: OutPoint,
+    /// The amount of sats associated with the above UTXO.
+    pub amount: u64,
+    /// The address where the newly minted sBTC will be deposited.
+    pub recipient: StacksAddress,
+}
+
+impl CompleteDepositV1 {
+    const CONTRACT_NAME: &'static str = "sbtc-deposit";
+    const FUNCTION_NAME: &'static str = "complete-deposit-wrapper";
+
+    /// Construct the input arguments to the complete-deposit-wrapper
+    /// contract call.
+    fn as_contract_args(&self) -> Vec<Value> {
+        let txid_data = self.outpoint.txid.to_byte_array().to_vec();
+        let txid = BuffData { data: txid_data };
+        let principle = StandardPrincipalData::from(self.recipient);
+
+        vec![
+            Value::Sequence(SequenceData::Buffer(txid)),
+            Value::UInt(self.outpoint.vout as u128),
+            Value::UInt(self.amount as u128),
+            Value::Principal(PrincipalData::Standard(principle)),
+        ]
+    }
+}
+
+impl AsContractCall for CompleteDepositV1 {
+    /// Converts this struct to a Stacks Contract call
+    fn as_contract_call(&self, deployer: StacksAddress) -> TransactionContractCall {
+        TransactionContractCall {
+            address: deployer,
+            // The following From::from calls are more dangerous than they
+            // appear. Under the hood they call their TryFrom::try_from
+            // implementation and then unwrap them(!). We check that this
+            // is fine in our test.
+            function_name: ClarityName::from(CompleteDepositV1::FUNCTION_NAME),
+            contract_name: ContractName::from(CompleteDepositV1::CONTRACT_NAME),
+            function_args: self.as_contract_args(),
+        }
+    }
+
+    /// The post conditions for the transaction. We do not enforce any
+    /// conditions here, since we trust this contract (we deployed it).
+    fn post_conditions(&self, _: StacksAddress) -> StacksTxPostConditions {
+        StacksTxPostConditions {
+            post_condition_mode: TransactionPostConditionMode::Allow,
+            post_conditions: Vec::new(),
+        }
+    }
+}
+
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use super::*;
+
+    #[test]
+    fn deposit_contract_call_creation() {
+        // This is to check that this function doesn't implicitly panic. If
+        // it doesn't panic now, it can never panic at runtime.
```

Comment:
Thanks for clarifying. I was going to ask what we were testing here in practice. I wonder if we should consider adding a placeholder integration test somewhere and reference an issue to implement it. wdyt?

Reply:
Yeah, that makes sense. I'll add the place holder, I was planning on using https://github.com/stacks-network/sbtc/issues/264 as the ticket.

Reply:
Eh, that ticket might not be the right vehicle for that. I'll open another one later if it's a bad fit.